web_scraper/chicago_scraper.py

```diff
@@ -0,0 +1,155 @@
+import pandas as pd
+import numpy as np
+import time
+import platform
+from dotenv import load_dotenv
+from selenium import webdriver
+from selenium.webdriver.chrome.service import Service
+from selenium.webdriver.support.wait import WebDriverWait
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support import expected_conditions
+import os
+from supabase import create_client, Client
+from datetime import datetime
+import pytz
+
+load_dotenv()
+
+url: str = os.environ.get("SUPABASE_URL")
+key: str = os.environ.get("SUPABASE_SR")
+
+supabase: Client = create_client(url, key)
+
+current_os = platform.system()
+
+if current_os == "Windows":
+    service = Service(executable_path="drivers/chromedriver.exe")  
+elif current_os == "Darwin": 
+    service = Service(executable_path="drivers/chromedriver")  
+else:
+    raise Exception("Unsupported operating system")
+
+options = webdriver.ChromeOptions()
+options.add_argument("--headless=new")
+
+driver = webdriver.Chrome(options=options, service=service)
+url = "https://webapps1.chicago.gov/licensedcontractors/all"
+
+driver.get(url)
+table = driver.find_element(By.ID, 'paginatedTable')
```

Comment:
What happens when we don't find this element or any of the others? We should consider the case that things change and perhaps wrapping things in try-except to handle the errors or cases that the html on the site changes. The exception can warn us somehow or handle the alternative. Alternatively you can see if you the web driver comes with fallback values when it can't find the tag and do and if statement